client/services/CodeStorageServiceSpec.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the CodeStorageService.
+ */
+
+describe('CodeStorageService', function() {
+  var CodeStorageService;
+  var questionIds = [];
+  var studentSubmittedCodes = [];
+  var i;
+  for (i = 0; i < 5; i++) {
+    questionIds[i] = 'question_' + (i + 1);
+    studentSubmittedCodes[i] = 'code_' + (i + 1);
+  }
+  const language = 'python';
+  const failedLanguage = 'java';
+
+  var generateRandomChars = function(number) {
+    var generatedChars = "";
+    var possible = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+
+    for (i = 0; i < number; i++) {
+      generatedChars += possible.charAt(
+        Math.floor(Math.random() * possible.length));
+    }
+    return generatedChars;
+  };
+
+  var randomQuestionIds = [];
```

Comment:
True, but you're doing a fair bit of extra effort which will likely confuse people reading your code after the fact when it's not super necessary. If your code depends on the question id, it might be worth mentioning in a comment, but you should be able to just use dummy or mock names for question ids here since you're not testing over actual question data.